atomisticparsers/h5md/parser.py

```diff
@@ -0,0 +1,690 @@
+        for key, sec_key in system_keys.items():
+            path = f'{self._path_group_particles_all}.{self._nomad_to_particles_group_map[key]}'
+            value = self._data_parser.get(path)
+            values_dict[sec_key][key] = get_value(value, steps, path=path)
+
+        # get the box quantities
+        box = self._data_parser.get(f'{self._path_group_particles_all}.box')
+        if box is not None:
+            box_attributes = {'dimension': 'system', 'periodic': 'system'}
+            for box_key, sec_key in box_attributes.items():
+                path = f'{self._path_group_particles_all}.box.{self._nomad_to_box_group_map[box_key]}'
+                value = self._data_parser.get(path, isattr=True)
+                values_dict[sec_key][box_key] = [value] * n_frames if value is not None else None
+            box_keys = {'lattice_vectors': 'system'}
+            for box_key, sec_key in box_keys.items():
+                path = f'{self._path_group_particles_all}.box.{self._nomad_to_box_group_map[box_key]}'
+                value = self._data_parser.get(path)
+                values_dict[sec_key][box_key] = get_value(value, steps, path=path)
+
+        # populate the dictionary
+        for i_step, step in enumerate(steps):
+            self._system_info['system'][step] = {key: val[i_step] for key,val in values_dict['system'].items()}
+            self._system_info['calculation'][step] = {key: val[i_step] for key,val in values_dict['calculation'].items()}
+
+    def parse_observable_info(self):
+        self._observable_info = {
+            'configurational': {},
+            'ensemble_average': {},
+            'correlation_function': {}
+        }
+        thermodynamics_steps = []
+        observables_group = self._data_parser.get('observables')
+        if observables_group is None:
+            return self._observable_info
+
+        def get_observable_paths(observable_group: Dict, current_path: str) -> List:
+            paths = []
+            for obs_key in observable_group.keys():
+                path = obs_key + '.'
+                observable = self._data_parser.get_value(observable_group, obs_key)
+                observable_type = self._data_parser.get_value(observable_group, obs_key).attrs.get('type')
+                if not observable_type:
+                    paths.extend(get_observable_paths(observable, f'{current_path}{path}'))
+                else:
+                    paths.append(current_path + path[:-1])
+
+            return paths
+
+        observable_paths = get_observable_paths(observables_group, current_path='')
+        for path in observable_paths:
+            full_path = f'observables.{path}'
+            observable = self._data_parser.get_value(observables_group, path)
+            observable_type = self._data_parser.get_value(observables_group, path).attrs.get('type')
+            observable_name = path.split('.')[0]
```

Comment:
changed to `observable_name, observable_label = path.split('.', 1) if len(path.split('.')) > 1 else [path, '']` for the case that path has no '.' e.g., path = 'test', which would throw an error for the suggested code

Reply:
ah yes sorry, then please do not do one line because this is not also pretty we split twice. do instead:
```python
name_label = path.split('.', 1)
observable_name = path[0]
observable_label = path[1] if len(path) > 1 else ''